Backend

chlouly opened the pull request:
Many small changes have been made and some big ones. Look through the commits if you'd like. I mostly just added some support for the RDS server. More should be coming soon.

Commit:
Modified npm scripts to run the server in various ways
The scripts found in './backend/package.json' have been modified to
run the server and modify the contents of the directory in some way.
The new scripts are as follows:
	* 'npm run clean': Removes all dependency files (found in the
		'./backend/node_modules' directory, as well as recur-
		sivelly removes all compiled javascript files (*.js).
	* 'npm run build': Reinstalls all dependencies and compiles
		all .ts sourcecode to .js.
	* 'npm run start': Runs the server code in app.ts.
	* 'npm run dev': Performs the first three scripts in succes-
		sion, essentially wiping the backend directory clean
		and starting the server over again.
	* 'npm run test': Currently fails as no tests have been
		written.

Run any one of the above 5 commands in your command line to yield
the described results.

Commit:
Refactored Monster and Player IDs to be numbers
This refactor was done so that the primary keys of the entries of the
AWS Postgres RDS server can be used as the ID values of each class. The
Player Class' Bench and Rosters were also modified so that they only hold
their monster's IDs and not the entire object to decrease the ammount
of memory that the system consumes. This comes with the drawback of having
to retrieve each monster from the RDS database and their images from the
S3 bucket when necessary, which will result in a slower server, however this
could be resolved by caching monster data strategically.

Commit:
Added makeshift ORMs to each type
Each class with its associated types and functions have been
separated out into their own files. Each of the tree classes now
has an ORM that is called {CLASSNAME}Row, an interface where the
data of the class can be placed so that it is readable by pg when
storing or fetching data from the database.

Commit:
Setting up the foundation for our database interaction
A new dependency has been added. pg is a package that allows the code
to interact with a postgres database server. We are using it to do just
that.

Additionally, there are two new files, one that contains functions used
to setup the database (./backend/rds_config.ts) and one that will soon
handle all of the storing, fetching, and deleting (./backend/rds_actions.ts)

There are also 5 new variables that are stored in the .env file:
	* RDS_USER= The Master Username of the server.
	* RDS_HOST= The endpoint that we will use as the server's address.
	* RDS_PASSWORD: The Master Password to access the server.
	* RDS_DB_NAME: The name of the DB on the RDS server.
	* RDS_PORT: The port on which our program will interact with the
		RDS server.

Commit:
Moving the game element class code
For the sake of repo clenliness, the three files ./backend/player.ts,
./backend/egg.ts and ./backend/monster.ts have all been moved into a new
directory. They can be found at: ./backend/game_elems.

Commit:
Unsaved changes - Sorry

Commit:
Bug fixes
Some small errors have been fixed. Also the .env variable PORT has
been changed to SERVER_PORT to avoid confusion.

RomanczuG commented:
Looks good!

RomanczuG reviewed (approved):
I like the progress and clean up. Good job

Commit:
Added a simple script to help with deployment on the EC2
The script found in backend/redeploy kills the current pm2 process,
recompiles the backend code and starts the pm2 daemon.

Note pm2 must be intalled globally for this script to work.

RomanczuG reviewed (approved):
Do you want me to set up the EC2 deployment GitHub action?
